gtpl/topic/hints.md

```diff
@@ -0,0 +1,44 @@
+
+1. Always run the tests. Usually running just `go test` in the exercise directory is all you need to do here, but if there are any special requirements for an exercise, you'll find them in the instructions for that exercise.
+
+    Any solution which does not pass the tests will be bounced straight back to you by the mentor, so to avoid this, do one final `go test` before submitting, just to make sure.
+
+    If you can't get all the tests to pass, and you've made a genuine attempt to fix the problem yourself, feel free to submit the solution anyway and ask for help. Mentors will be happy to give you hints if you can show them that you've put some effort into your solution.
+
+2. Always ensure your code is formatted with `gofmt`. Most editors that support Go can be configured to do this automatically. If you're coming to Go from other languages, you may not be used to the idea that there's one, and only one, accepted way to format Go code, and it's the `gofmt` way. At first this may seem overly restrictive, but there are great advantages to standard formatting, not least that it avoids a lot of futile arguments about which is the best way to format code!
+
+    Again, mentors will expect you to `gofmt` your code, and anything which is not correctly formatted will be bounced back to you. Don't waste the mentors' time and patience by arguing with them about this. If you want to join the worldwide community of Gophers, that means accepting community standards, many of which may seem silly or pointless to you at first. Stick with it; you may feel differently about things further down the line.
+
+3. Always make sure your code passes `golint`. `golint` is a tool that analyses your code for common errors and problems, and also enforces things like documentation comments for your functions and identifiers. Again, your editor can usually lint your code automatically, and it's a good idea to set this up. Run `go get -u golang.org/x/lint/golint` to install `golint`, and check your code with it before submitting.
+
+    Like all automated tools, `golint` is not perfect, and there are occasions where you can safely ignore its advice, but you won't encounter any of them in Exercism exercises. Therefore, please follow all `golint`'s suggestions for now. Some of them may seem silly or excessive, but again, stick with it for now. You are building habits which will stand you in good stead later.
+
+    One of the first things `golint` is likely to complain about is missing documentation comments on your functions. Every Go function should have a comment above it explaining what the function does, what arguments it takes, and what it returns. Don't skip this or write silly comments to silence `golint`, like "Square yada yada yada." Sure, it's just an exercise, and you don't really need documentation, but you're learning the skills and habits to be a successful Go programmer, and good documentation is one of them.
```

Comment:
I'd probably mention somewhere that this is only about exported functions. Otherwise students might freak ;)

Reply:
Yes, I did consider that. But I think every Go function _should_ have a documentation comment. So I'm inclined to let that point stand.